Sources/Commands/SwiftTool.swift

```diff
@@ -594,6 +594,33 @@ public class SwiftTool {
             throw error
         }
     }
+    
+    /// Evaluate extensions for any reachable targets in the graph, and return a mapping from targets to corresponding evaluation results.
+    func evaluateExtensions(graph: PackageGraph) throws -> [ResolvedTarget: [ExtensionEvaluationResult]] {
```

Comment:
Every build system (of any IDE) that wants to incorporate libSwiftPM support would need to do something like this, where after loading the package graph, it evaluates the package extensions in the graph and gets back a mapping from targets to the ordered list of results of applying the target's specified extensions to that target.  Those results are build-system neutral descriptions of command that then need to be incorporated into the build system as appropriate.